synapse/handlers/user_directory.py

```diff
@@ -591,8 +591,29 @@ async def _unsafe_refresh_remote_profiles(self) -> None:
         )
 
         if not servers_to_refresh:
-            # TODO Do we have any backing-off servers that we should try again
-            #      for eventually?
+            # Do we have any backing-off servers that we should try again
+            # for eventually?
+            end_of_time = 1 << 62
+            backing_off_servers = (
+                await self.store.get_remote_servers_with_profiles_to_refresh(
+                    now_ts=end_of_time, limit=1
+                )
```

Comment:
I am curious why you are passing in `end_of_time` (which I presume is a Very Large Number) here to `now_ts`?

Reply:
it's a fair question that would be worth a comment.

But to answer: these functions return remote servers or users that are ready to be refreshed *now*. By giving them a time far in the future, we can ask for the next server/user to be refreshed.